### PR TITLE
SCALE instrinsic lowering

### PR DESCRIPTION
**Pull request by AlexisPerry**
- added basic implementation for lowering of the SCALE intrinsic
- added tests for lowering of SCALE intrinsic

### CODE REVIEW

File: flang/lib/Lower/NumericRuntime.cpp [hunk 1]
```diff
@@ -93,6 +117,33 @@ Fortran::lower::genRRSpacing(Fortran::lower::FirOpBuilder &builder,
...
+  else if (fltTy.isF128())
+    func = Fortran::lower::getRuntimeFunc<ForcedScale16>(loc, builder);
+  else
+    TODO(loc, "unsupported real kind in Scale lowering");
+
+  auto funcTy = func.getType();
+  llvm::SmallVector<mlir::Value> args = {
+    builder.createConvert(loc, funcTy.getInput(0), x),
+    builder.createConvert(loc, funcTy.getInput(1), i)
+  };
```
Comment (schweitzpgi) [inline flang/lib/Lower/NumericRuntime.cpp:142]: Could this be simplified with `Fortran::lower::createArguments`?

**Review by schweitzpgi — COMMENTED** [review]
Looks good. Thank you, Alexis.

File: flang/lib/Lower/NumericRuntime.cpp [hunk 2]
```diff
@@ -93,6 +117,33 @@ Fortran::lower::genRRSpacing(Fortran::lower::FirOpBuilder &builder,
...
+  if (fltTy.isF32())
+    func = Fortran::lower::getRuntimeFunc<mkRTKey(Scale4)>(loc, builder);
+  else if (fltTy.isF64())
+    func = Fortran::lower::getRuntimeFunc<mkRTKey(Scale8)>(loc, builder);
+  else if (fltTy.isF80())
+    func = Fortran::lower::getRuntimeFunc<ForcedScale10>(loc, builder);
+  else if (fltTy.isF128())
+    func = Fortran::lower::getRuntimeFunc<ForcedScale16>(loc, builder);
+  else
+    TODO(loc, "unsupported real kind in Scale lowering");
```
[the remark this answers is not in the document]
Comment (AlexisPerry) [reply flang/lib/Lower/NumericRuntime.cpp:136]: I don't know of other types that need to be handled, so I'll go ahead and make the change.

**Review by psteinfeld — CHANGES_REQUESTED** [review]
Thanks for working on this!

I don't have anything to add to Eric's comments.

Please run clang-format on all of the .h and .cpp source files, though.

**Comment by AlexisPerry** [comment]
> Please run clang-format on all of the .h and .cpp source files, though.

@psteinfeld Thanks for catching that, I totally spaced it.  Should be fixed now.

**Review by psteinfeld — APPROVED** [review]
All builds, tests, and looks good.

Thanks, Alexis!

**Pull request merged**